src/lib.rs

```diff
@@ -118,27 +118,26 @@ impl Default for Settings {
     }
 }
 
-impl<Spi, CommsError, Output, Input, PinError, Delay>
-    Sx127x<SpiWrapper<Spi, CommsError, Output, Input, PinError, Delay>, CommsError, PinError>
+impl<Spi, SpiError, CsPin, BusyPin, ReadyPin, ResetPin, PinError, Delay>
+    Sx127x<SpiWrapper<Spi, SpiError, CsPin, BusyPin, ReadyPin, ResetPin, PinError, Delay>, SpiError, PinError>
 where
-    Spi: Transfer<u8, Error = CommsError> + Write<u8, Error = CommsError>,
-    Output: OutputPin<Error = PinError>,
-    Input: InputPin<Error = PinError>,
-    Delay: delay::DelayMs<u32>,
+    Spi: Transfer<u8, Error = SpiError> + Write<u8, Error = SpiError>,
+    CsPin: OutputPin<Error = PinError>,
```

Comment:
you got most of the way! need to include `InputPin` and a couple more trait bounds to make it all work.

```suggestion
    CsPin: OutputPin<Error = PinError>,
    BusyPin: InputPin<Error = PinError>,
    ResetPin: OutputPin<Error = PinError>,
```

Reply:
Thank you!